clang/lib/Sema/SemaExpr.cpp

```diff
@@ -2347,6 +2347,27 @@ Sema::BuildDeclRefExpr(ValueDecl *D, QualType Ty, ExprValueKind VK,
   return E;
 }
 
+// Check whether a similar function-like macro exists and suggest it
+static bool isFunctionLikeMacro(const DeclarationName &Name, Sema &SemaRef,
+                                const SourceLocation &TypoLoc) {
+
+  if (IdentifierInfo *II = Name.getAsIdentifierInfo()) {
+    if (II->hasMacroDefinition()) {
+      MacroInfo *MI = SemaRef.PP.getMacroInfo(II);
+      if (MI && MI->isFunctionLike()) {
+        SemaRef.Diag(TypoLoc,
+                     diag::err_undeclared_var_use_suggest_func_like_macro)
+            << II->getName();
```

Comment:
```suggestion
            << II;
```
This will ensure the name is properly quoted by the diagnostics engine.

Reply:
Tried it, but gives `'FOO1' is defined as an object-like macro; did you mean ''FOO1'(...)'?` so reverted back.

Reply:
Ah, sorry! I didn't pick up that it was reusing `%0`, so reverting back makes sense.